src/Illuminate/Console/Command.php

```diff
@@ -189,7 +198,94 @@ public function run(InputInterface $input, OutputInterface $output)
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        return $this->laravel->call([$this, 'handle']);
+        if ($this->hasValidInput()) {
+            return $this->laravel->call([$this, 'handle']);
+        }
+
+        return $this->displayValidationErrors();
+    }
+
+    /**
+     * Check if the argument and option values are valid.
+     *
+     * @return bool
+     */
+    protected function hasValidInput()
+    {
+        if (empty($rules = $this->rules())) {
+            return true;
+        }
+
+        try {
+            $this->validator = $this->validator ?? $this->makeValidator();
+        } catch (BindingResolutionException $e) {
+            return true;
+        }
+
+        $this->validator->make(
+            array_merge($this->arguments(), $this->options()),
+            $rules,
+            $this->messages(),
+            $this->attributes()
+        );
+
+        return ! $this->validator->fails();
```

Comment:
@surgiie Does this code work as intended? I haven't tested it, but just looking at it it seems wrong.

`$this->validator = $this->validator ?? $this->makeValidator();`

After this line `$this->validator` will be an instance of a validator factory and then on this line you are calling the `fails` method on it which is a validator, not a validator factory method.

There is a call above to `$this->validator->make()` which creates a validator instance, but the result isn't assigned to any variable.

Am I missing something?

Reply:
@X-Coder264 Youre right. Ive corrected this and updated the name of the function to be `getValidationFactory` which is a bit less confusing and tested this change with it working as intended now. Thanks for catching this!  In a previous commit i was using an actual validator instance before switching to the factory/contract so thats where i missed this issue when i made the quick change.

Reply:
@surgiie Great, but I think that this revealed a need for some tests for this to exist to prevent a regression of this kind in the future and to ensure that this works as expected.